src/qibo/transpiler/optimizer.py

```diff
@@ -66,3 +67,261 @@ def __call__(self, circuit: Circuit):
             else:
                 new.add(fgate)
         return new
+
+
+class InverseCancellation(Optimizer):
+    """
+    Identifies and removes pairs of adjacent gates from a quantum circuit.
+    """
+
+    def __init__(self):
+        self.inverse_gates = (
+            gates.H | gates.Y | gates.Z | gates.X | gates.CNOT | gates.CZ | gates.SWAP
+        )
+        self.pos_rm_inv_gate = []
```

Comment:
Move the class docstrings to the `__init__` method and add any `Args` when necessary. This and the other classes.